atp/client.go

```diff
@@ -199,23 +203,20 @@ func (c *client) Execute(
 	}
 	c.logger.Debugf("Step '%s' started, waiting for response...", stepData.ID)
 
-	defer c.handleStepComplete(stepData.RunID, receivedSignals)
+	if receivedSignals != nil {
+		defer c.handleStepComplete(stepData.RunID)
+	}
 	return c.getResult(stepData, cborReader)
 }
 
 // handleStepComplete is the deferred function that will handle closing of the received channel.
-func (c *client) handleStepComplete(runID string, receivedSignals chan schema.Input) {
-	if receivedSignals != nil {
-		c.logger.Infof("Closing signal channel for finished step")
-		// Remove from the map to ensure that the client.Close() method doesn't double-close it
-		c.mutex.Lock()
-		// Validate that it exists, since Close() could have been called early.
-		_, exists := c.runningSignalReceiveLoops[runID]
-		if exists {
-			delete(c.runningSignalReceiveLoops, runID)
-		}
-		c.mutex.Unlock()
-	}
+func (c *client) handleStepComplete(runID string) {
+	c.logger.Infof("Closing signal channel for finished step")
+	// Remove from the map to ensure that the client.Close() method doesn't double-close it
+	c.mutex.Lock()
+	// Validate that it exists, since Close() could have been called early.
```

Comment:
This comment is no longer strictly accurate, since we're not "validating", but simply relying on `delete` on an undefined key being a no-op.

Reply:
Quite right...I should have caught that.

I removed the comment at line 217, and reworked the function header comment.

I also removed the comment at line 215, because I can't make sense of it anymore:  I don't see where `client.Close()` touches `runningSignalReceiveLoops` or the `receivedSignals` channel.  (@jaredoconnell?)